Taskfile.dist.yaml

```diff
@@ -16,15 +37,239 @@ tasks:
         done
       - git signatures push {{.CLI_ARGS}}
 
+  _build:cgo:dev: &build_cgo_dev
+    dir: "{{.dir}}"
+    cmds:
+      - go build {{if eq .raceDetectorEnabled "true" }}-race{{end}} -o "{{.outputDir | default (printf "./build/%s-%s" .targetOS .targetArch)}}/trdl{{if (eq .targetOS "windows")}}.exe{{end}}"
+        -compiler gc -ldflags="{{.cgoLDFlags}}" -tags="{{.cgoTags}}" {{.extraGoBuildArgs}} {{.CLI_ARGS}} {{.pkg | default .package}}
+    env:
+      CGO_ENABLED: "1"
+      GOOS: "{{.targetOS}}"
+      GOARCH: "{{.targetArch}}"
+      CC: "{{.cc}}"
+    vars:
+      raceDetectorEnabled: '{{.raceDetectorEnabled | default "false"}}'
+      dir: '{{.dir | default "client"}}'
+
+  _build:cgo:dist:
+    <<: *build_cgo_dev
+    cmds:
+      - go build -o "{{.outputDir | default (printf "./dist/%s/%s-%s/bin" .version .targetOS .targetArch)}}/trdl{{if (eq .targetOS "windows")}}.exe{{end}}"
+        -compiler gc -ldflags="{{.cgoLDFlags}}" -tags="{{.cgoTags}}" {{.extraGoBuildArgs}} {{.CLI_ARGS}} {{.pkg | default .package}}
+
+  _build:go:dev: &build_go_dev
+    dir: "{{.dir}}"
+    cmds:
+      - go build {{if eq .raceDetectorEnabled "true" }}-race{{end}} -o "{{.outputDir | default (printf "./build/%s-%s" .targetOS .targetArch)}}/trdl{{if (eq .targetOS "windows")}}.exe{{end}}"
+        -ldflags="{{.goLDFlags}}" -tags="{{.goTags}}" {{.extraGoBuildArgs}} {{.CLI_ARGS}} {{.pkg | default .package}}
+    env:
+      CGO_ENABLED: "0"
+      GOOS: "{{.targetOS}}"
+      GOARCH: "{{.targetArch}}"
+    vars:
+      raceDetectorEnabled: '{{.raceDetectorEnabled | default "false"}}'
+      dir: '{{.dir | default "client"}}'
+
+  _build:go:dist:
+    <<: *build_go_dev
+    cmds:
+      - go build -o "{{.outputDir | default (printf "./dist/%s/%s-%s/bin" .version .targetOS .targetArch)}}/trdl{{if (eq .targetOS "windows")}}.exe{{end}}"
+        -ldflags="{{.goLDFlags}}" -tags="{{.goTags}}" {{.extraGoBuildArgs}} {{.CLI_ARGS}} {{.pkg | default .package}}
+
+  build:
+    desc: "Build all dev binaries."
+    run: once
+    cmds:
+      - task: build:dev:{{.os}}:{{.arch}}
+        vars:
+          outputDir: '{{.outputDir | default "../bin"}}'
+          extraGoBuildArgs: "{{.extraGoBuildArgs}}"
+          raceDetectorEnabled: '{{.raceDetectorEnabled | default "false"}}'
+          dir: '{{.dir | default "client"}}'
+      - task: build:vault-plugin
+        vars:
+          dir: '{{.dir | default "server"}}'
```

Comment:
This needs to be reorganized, there should be 2 tasks:
1. build:client:dev:{{.os}}:{{.arch}}
2. build:server:dev:{{.os}}:{{.arch}}

Reply:
no dir variable needed